packages/playwright/src/worker/workerMain.ts

```diff
@@ -430,15 +430,25 @@ export class WorkerMain extends ProcessRunner {
         throw firstAfterHooksError;
     }).catch(() => {});  // Ignore the top-level error, it is already inside TestInfo.errors.
 
-    // Create warning if any of the async calls were not awaited in various stages.
-    if (!process.env.PW_DISABLE_FLOATING_PROMISES_WARNING && testInfo._floatingPromiseScope.hasFloatingPromises()) {
-      testInfo.annotations.push({ type: 'warning', description: `Some async calls were not awaited by the end of the test. This can cause flakiness.` });
-      testInfo._floatingPromiseScope.clear();
-    }
-
-    if (testInfo._isFailure())
+    if (testInfo._isFailure()) {
       this._isStopped = true;
 
+      // Only if failed, create warning if any of the async calls were not awaited in various stages.
+      if (!process.env.PW_DISABLE_FLOATING_PROMISES_WARNING && testInfo._floatingPromiseScope.hasFloatingPromises()) {
```

Comment:
Somewhat unrelated to this PR, but we usually name environment variables intended for the end user as `PLAYWRIGHT_...`, not `PW_...`.

Reply:
Good point, but this is already public API. Leaving for now.